crates/gitbutler-repo/src/repository_ext.rs

```diff
@@ -394,127 +397,128 @@ impl RepositoryExt for git2::Repository {
     fn sign_buffer(&self, buffer: &[u8]) -> Result<BString> {
         // check git config for gpg.signingkey
         // TODO: support gpg.ssh.defaultKeyCommand to get the signing key if this value doesn't exist
-        let signing_key = self.config()?.get_string("user.signingkey");
-        if let Ok(signing_key) = signing_key {
-            let sign_format = self.config()?.get_string("gpg.format");
-            let is_ssh = if let Ok(sign_format) = sign_format {
-                sign_format == "ssh"
-            } else {
-                false
-            };
+        let repo = gix::open(self.path())?;
+        let config = repo.config_snapshot();
+        let signing_key = config.string("user.signingkey");
+        let Some(signing_key) = signing_key else {
+            return Err(anyhow::anyhow!("No signing key found"));
```

Comment:
There is a `bail!` macro which is equivalent to `return Err(anyhow::anyhow!(...))`

Reply:
Good catch, done!